### PR TITLE
Unifying calls to Github + Handling rejectedPromises properly

### LINKED ISSUE #175: HttpError: Resource not accessible by integration silently failing Herald comment actions
## Description

In a test PR, I made a change expecting Herald to post a comment. Unfortunately, it didn't because of the following error:
```
(node:3080) UnhandledPromiseRejectionWarning: HttpError: Resource not accessible by integration
    at /home/runner/work/use-herald-action/use-herald-action/dist/index.js:22140:23
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async run (/home/runner/work/use-herald-action/use-herald-action/dist/index.js:4383:29)
(node:3080) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:3080) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Here's the job: https://github.com/gagoar/use-herald-action/pull/174/checks?check_run_id=1224140598

## Context

Worth noting that this isn't the first time this has happened. In an earlier PR ([failed job](https://github.com/gagoar/use-herald-action/runs/1140973417?check_suite_focus=true)), this error was thrown but not detected since the changes in said PR did not warrant a Herald comment.

Other projects are experiencing the same issue. See links below for a better explanation on why this is happening

Relevant issues in other projects:
https://github.com/actions/first-interaction/issues/10
https://github.com/actions/labeler/issues/12

## Recommendations

Using the app token may solve the permissions issue that causes this error.

## Actionable

Better error handling - we need the action to explode and report back with an ❌  when this happens.

### PR DESCRIPTION
**Pull request by gagoar**
**Issue Reference**: fixes #175 

## Description

Fixing action so it fails when GitHub fails. 

As issue #175 explains we weren't handling when a  Github request will get rejected, this happens for several reasons, but more notably when the token used doesn't have enough permissions. 

I've also unified the calls when possible, so we don't call Github that much. 

- Unified labels to call once per run.
- Unified assignees to call once per run. 
- Unified reviewers to call once per run. 

I've removed the P-queue from  `comments` but I will add it again before merging. 

## Motivation and Context

It is important that the action fails when something like an invalid token is not able to perform the actions that is meant to perform. 

## How Has This Been Tested?
Unit tests!

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -131,7 +131,7 @@ export const main = async (): Promise<void> => {
 
           debug('groupNames', groupNames);
 
-          await Promise.all([
+          await Promise.all(
```
Comment (gagoar) [inline src/index.ts:134]: this was the main problem. the array was wrapping other arrays thus not allowing Promise.all to really catch anything. took me a min to realize. 🔫

**Comment by gagoar** [comment]
<details open>

   <summary> Hi there, given these changes, Herald suggest these users should take a look! </summary>

   | Rule                                                                                                                                                                                            | Mention |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :-----: |
| [This is a rule to attach @gagoar when workflow changes](https://github.com/gagoar/use-herald-action/blob/fb8e0736363c146d86f23ad0229b1b6fd2118193/herald_rules/add_comment_for_workflows.json) | @gagoar |

  </details>

File: src/index.ts [hunk 2]
```diff
@@ -147,8 +147,11 @@ export const main = async (): Promise<void> => {
               };
 
               return action(client, options);
-            }),
-          ]);
+            })
+          ).catch((error: Error) => {
+            debug('We found and error calling GitHub:', error);
```
Comment (cyamonide) [inline src/index.ts:152]:
```suggestion
            debug('We found an error calling GitHub:', error);
```

**Review by cyamonide — APPROVED** [review]
looks good! one typo

**Pull request merged**